src/indra_cogex/sources/sider/__init__.py

```diff
@@ -0,0 +1,257 @@
+from indra.databases.identifiers import get_ns_id_from_identifiers
+from indra_cogex.constants import MODULE
+from indra_cogex.representation import Node, Relation, standardize
+from indra_cogex.sources import Processor
+
+VERSION = "4.1"
+SUBMODULE = MODULE.submodule("indra", "cogex", "sider", VERSION)
+URL = "http://sideeffects.embl.de/media/download/meddra_all_se.tsv.gz"
+SIDE_EFFECTS_HEADER = [
+    # 'STITCH_FLAT_ID',
+    "STITCH_STEREO_ID",
+    # 'UMLS CUI from Label',
+    "MedDRA Concept Type",
+    "UMLS CUI from MedDRA",
+    "MedDRA Concept name",
+]
+
+
+def stitch_stereo_to_pubchem(cid: str) -> str:
+    assert cid.startswith("CID")
+    return str(cid[3:])
+
+
+class UmlsMapper:
+    """A utility class for mapping out of UMLS."""
+
+    prefixes = ["doid", "mesh", "hp", "efo", "mondo"]
+
+    def __init__(self):
+        """Prepare the UMLS mappings from PyOBO and Biomappings."""
+        #: A dictionary from external prefix to UMLS id to external ID
+        self.xrefs = {}
+
+        for prefix in self.prefixes:
+            self.xrefs[prefix] = {}
+            # Get external to UMLS
+            for external_id, umls_id in pyobo.get_filtered_xrefs(
+                prefix, "umls"
+            ).items():
+                self.xrefs[prefix][umls_id] = external_id
+            # Get UMLS to external
+            for umls_id, external_id in pyobo.get_filtered_xrefs(
+                "umls", prefix
+            ).items():
+                self.xrefs[prefix][umls_id] = external_id
+
+        # Get manually curated UMLS mappings from biomappings
+        biomappings_from_umls, biomappings_to_umls = Counter(), Counter()
+        for mapping in load_mappings():
+            if mapping["source prefix"] == "umls":
+                target_prefix = mapping["target prefix"]
+                biomappings_from_umls[target_prefix] += 1
+                target_id = mapping["target identifier"]
+                source_id = mapping["source identifier"]
+                if target_prefix in self.xrefs:
+                    self.xrefs[target_prefix][target_id] = source_id
+                else:
+                    self.xrefs[target_prefix] = {
+                        target_id: source_id,
+                    }
+            elif mapping["target prefix"] == "umls":
+                source_prefix = mapping["source prefix"]
+                biomappings_to_umls[source_prefix] += 1
+                source_id = mapping["source identifier"]
+                target_id = mapping["target identifier"]
+                if source_prefix in self.xrefs:
+                    self.xrefs[source_prefix][source_id] = target_id
+                else:
+                    self.xrefs[source_prefix] = {
+                        source_id: target_id,
+                    }
+
+        print("Mapping out of UMLS")
+        print(tabulate(biomappings_from_umls.most_common()))
+        print("Mapping into UMLS")
+        print(tabulate(biomappings_to_umls.most_common()))
```

Comment:
An alternative approach here could be to propagate these mappings into INDRA (see https://github.com/sorgerlab/indra/blob/master/indra/resources/biomappings.tsv) and add them to the BioOntology there, then the usual standardization procedure would take care of the necessary mappings. What do you think?